admin/src/core/services/cognito.service.ts

```diff
@@ -167,10 +132,90 @@ export class CognitoService {
+    this.awsCognitoConfig = await lastValueFrom(
+      this.http.get(url, { observe: "body", responseType: "json" })
+    ) as AwsCognitoConfig;
+
+    const amplifyConfig = this.toAmplifyConfig(this.awsCognitoConfig);
+    console.log("Using amplify config = " + JSON.stringify(amplifyConfig));
+    Amplify.configure(amplifyConfig);
+  }
+
+  private toAmplifyConfig(awsCognitoConfig: AwsCognitoConfig): ResourcesConfig {
+    // conversion to config aws-amplify (gen 2)
+    return {
+      Auth: {
+        Cognito: {
+          userPoolClientId: awsCognitoConfig.aws_user_pools_web_client_id,
+          userPoolId: awsCognitoConfig.aws_user_pools_id,
+          loginWith: {
+            oauth: {
+              domain: awsCognitoConfig.oauth.domain,
+              scopes: ['openid'],
+              redirectSignIn: [awsCognitoConfig.oauth.redirectSignIn],
+              redirectSignOut: [this.awsCognitoConfig.oauth.redirectSignOut],
+              responseType: 'code',
+            },
+          }
+        }
+      }
+    }
+  }
+
+  private getParameterByName(name) {
+    const url = window.location.href;
+    const regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)");
+    const results = regex.exec(url);
+    if (!results) {
+      return null;
+    }
+    if (!results[2]) {
+      return "";
+    }
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+  }
+
+  /**
+   * See OIDC Attribute Mapping mapping reference:
+   *      https://github.com/bcgov/nr-forests-access-management/wiki/OIDC-Attribute-Mapping
+   * The display username is "custom:idp_username" from token.
+   */
+  private parseToken(authToken: CognitoUserSession): CognitoAuthToken {
+    console.log("authToken: ", authToken)
+    const decodedIdToken = authToken.getIdToken().decodePayload();
+    const decodedAccessToken = authToken.getAccessToken().decodePayload();
+    return {
+      id_token: decodedIdToken,
+      access_token: decodedAccessToken,
+      jwtToken: authToken
+    };
+  }
+
+  /**
+   * Note:
+   * `aws-amplify` (v6) does not expose previous CognitoUserSession (JWT Token) and it does not
+   * expose 'refreshToken' anymore than previous v5.
+   * To get entire encoded token, calls to the 'toString()' is necessary (but is not documented 
+   * on aws-amplify).
+   * @returns newly fetched session converted into Promise<CognitoUserSession> type.
+   */
+  private async refreshAndObtainAwsCognitoUserSession(): Promise<CognitoUserSession> {
+    const authSession = await fetchAuthSession({ forceRefresh: true });
+    const cognitoUserSession = new CognitoUserSession( {
+      IdToken: new CognitoIdToken({IdToken: authSession.tokens.idToken.toString()}),
+      AccessToken: new CognitoAccessToken({AccessToken: authSession.tokens.accessToken.toString()}),
+      // RefreshToken: /* aws-amplify v6 does not provide refreshToken anymore, only internally for aws-amplify*/
+    })
```

Comment:
Previously from Amplify v5, 'CognitoUserSession' was the class returned from Amplify; in v6, there is no more. Here is the conversion to the 'CognitoUserSession' class (from lib `amazon-cognito-identity-js`) to still make FOM old code works without impacting backend.